include/triton/Dialect/TritonGPU/Transforms/Partition.h

```diff
@@ -121,6 +122,8 @@ bool hasPartition(Operation *op);
 // Annotate the op with the partition index or indices, and add the op
 // to the partitions it belongs to.
 void setPartition(Operation *op, Partition *partition);
+// Unset partition annotation from the op
+void unsetPartition(Operation *op);
 void setPartition(Operation *op, const SetVector<Partition *> &partitions);
 // Annotate the op with the partition indices. It should only be used in a pass
 // which does not work with Partition instances and iterate* functions, since
```

Comment:
Can we drop `optional` from

```
std::optional<SetVector<int>> getPartitionIds(Operation *op);
```
now ?

Reply:
we cannot yet do that, there is a lot of code in partition-scheduler relying on this to be optional via `hasPartition` helper.

Reply:
yeah, on an attempt it appears to be quite an invasive change that I prefer to do in a separate PR. I can take that as one of my next tasks.